builtin/k8s/platform.go

```diff
@@ -296,318 +294,392 @@ func (p *Platform) resourceDeploymentStatus(
+	deployConfig *component.DeploymentConfig,
+	ui terminal.UI,
+	result *Deployment,
+	state *Resource_Deployment,
+	csinfo *clientsetInfo,
+	sg terminal.StepGroup,
+) error {
+	// Prepare our namespace and override if set.
+	ns := csinfo.Namespace
+	if p.config.Namespace != "" {
+		ns = p.config.Namespace
 	}
 
-	// Update the deployment with our spec
-	deployment.Spec.Template.Spec = corev1.PodSpec{
-		Containers: []corev1.Container{container},
+	step := sg.Add("")
+	defer func() { step.Abort() }()
+	step.Update("Kubernetes client connected to %s with namespace %s", csinfo.Config.Host, ns)
+	step.Done()
+
+	step = sg.Add("Preparing deployment...")
+
+	clientSet := csinfo.Clientset
+	deployClient := clientSet.AppsV1().Deployments(ns)
+
+	// Determine if we have a deployment that we manage already
+	create := false
+	deployment, err := deployClient.Get(ctx, result.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		deployment = result.newDeployment(result.Name)
+		create = true
+		err = nil
+	}
+	if err != nil {
+		return err
 	}
 
-	// Override the default TCP socket checks if we have a probe path
-	if p.config.ProbePath != "" {
-		deployment.Spec.Template.Spec.Containers[0].LivenessProbe = &corev1.Probe{
-			Handler: corev1.Handler{
-				HTTPGet: &corev1.HTTPGetAction{
-					Path: p.config.ProbePath,
-					Port: intstr.FromInt(defaultPort),
-				},
-			},
-			InitialDelaySeconds: initialDelaySeconds,
-			TimeoutSeconds:      timeoutSeconds,
-			FailureThreshold:    failureThreshold,
-		}
+	// Setup our port configuration
+	if p.config.ServicePort == 0 && len(p.config.Ports) == 0 {
+		// nothing defined, set up the defaults
+		p.config.Ports = append(p.config.Ports, &Port{Port: DefaultServicePort, Name: "http"})
+	} else if p.config.ServicePort > 0 && len(p.config.Ports) == 0 {
+		// old ServicePort var is used, so set it up in our Ports map to be used
+		p.config.Ports = append(p.config.Ports, &Port{Port: p.config.ServicePort, Name: "http"})
+	} else if p.config.ServicePort > 0 && len(p.config.Ports) > 0 {
+		// both defined, this is an error
+		return fmt.Errorf("cannot define both 'service_port' and 'ports'. Use" +
+			" 'ports' for configuring multiple container ports")
+	}
 
-		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe = &corev1.Probe{
-			Handler: corev1.Handler{
-				HTTPGet: &corev1.HTTPGetAction{
-					Path: p.config.ProbePath,
-					Port: intstr.FromInt(defaultPort),
+	envVars := make(map[string]string)
+	// Add deploy config environment to container env vars
+	for k, v := range p.config.StaticEnvVars {
+		envVars[k] = v
+	}
+	for k, v := range deployConfig.Env() {
+		envVars[k] = v
+	}
+
+	// Check autoscaling
+	if p.config.AutoscaleConfig != nil && p.config.CPU == nil {
+		ui.Output("For autoscaling in Kubernetes to work, a deployment must specify "+
+			"cpu resource limits and requests. Otherwise the metrics-server will not properly be able "+
+			"to scale your deployment.", terminal.WithWarningStyle())
+	}
```

Comment:
This can still be set through the `resource_limits` and `resource_requests` map, so I'd recommend removing this and putting it back where we create the container